loki/bulk/scheduler.py

```diff
@@ -448,7 +457,12 @@ def _get_definition_items(_item, sgraph_items):
 
         if transformation.creates_items:
             self._discover()
+            # Rebuild after discovery
+            self._sgraph = SGraph.from_seed(self.seeds, self.item_factory, self.config)
+
             self._parse_items()
+            # And rebuild after parsing
+            self._sgraph = SGraph.from_seed(self.seeds, self.item_factory, self.config)
```

Comment:
Given that `_discover()` and `_parse_items()` are now the two patterns which mandate a rebuild, I'm wondering if the `SGraph` construction and caching should not just be put in there?